### ISSUE TITLE
(Willing to PR) Will it be welcomed if speeding up algorithms like PPO and code refactor/cleanup?

### ISSUE THREAD

**Issue by somedude5194** [somedude5194/trl#2]
### Method description

Feature request
Hi thanks for the PR! I wonder whether it will be welcomed (e.g. PR gets reviewed and merged) if PRs are made to speed up algorithms such as PPO and RLOO?

For speedup, for example:

Use SGLang to do rollout (generation) phase instead of vanilla HuggingFace model / DeepSpeed model. This seems to speed up generation a lot.
Delete ref & reward model from GPU when they are not used, such that gpu memory is not occupied, and can support larger batch size or larger models.
...
For refactor/cleanup, just standard tiny things like:

Extracting functions to avoid very long single function
Extract logits of metric calculations to make main function focused on core logic
Try to merge RLOO with PPO into same class (not checked in details, may or may not do this)
...
In addition, I would appreciate it if I could know some comparisons between TRL and OpenRLHF/verl/DeepSpeed-Chat.

Motivation
(see above)

Your contribution
yes

### Open source status

- [ ] The method implementation is available
- [ ] The model weights are available
- [ ] The training datasets are available
